### DISCUSSION TITLE
[Chapter 04] HTTP 메서드의 멱등성과 브라우저 캐시의 관계

### DISCUSSION THREAD

**Discussion by JoisFe** [cowaine/spring-boot-for-msa#29] [Chapter 04. Spring Web MVC Overview] [answered]
브라우저에서 웹페이지를 보여줄때 효율성을 위해 캐시에 저장해둔 페이지를 반환해주는 것으로 압니다. 멱등성을 가지는 메서드 요청에 대해서 페이지가 변화에 큰 영향을 끼치지 않을 것으로 보이는데 이들이 관련된 것이 있는지 궁금합니다.

**Comment by corock** [comment, marked as answer]
HTTP 공식 사이트가 있는지는 모르겠는데 MDN Web Docs 에서는 [HTTP 요청 메서드](https://developer.mozilla.org/ko/docs/Web/HTTP/Methods) 페이지에 메서드 별로 명시되어 있는 3요소가 있습니다.

1. 안정성(_Safe_): 리소스를 수정하지 않는가?
2. 멱등성(_Idempotent_): 여러 번 요청하더라도 요청 결과가 같은가?
3. 캐시 가능성(_Cachable_): 재사용을 위해 응답을 저장할 수 있는가?

찾아보니 HTTP 메서드 9가지 중 자주 사용하는 5개를 보면 GET 메서드만 안정성이 보장되며, 멱등성은 POST, PATCH 를 제외하고 보장하네요.
캐시가 가능한 메서드는 PUT 을 제외하고 모두 가능하긴 하지만, 캐시 키를 가지고 구현하는 과정이 복잡하여 실무에서는 GET 정도만 사용한다고 합니다.

**Reply by JoisFe** [reply]
https://velog.io/@msung99/HTTP-%EB%A9%94%EC%86%8C%EB%93%9C%EC%9D%98-%EC%86%8D%EC%84%B1-%EC%95%88%EC%A0%84-%EB%A9%B1%EB%93%B1Idempotent-%EC%BA%90%EC%8B%9CCacheable

참고!
